src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/MapCode/MapCodeEndpoint.cs

```diff
@@ -51,22 +54,30 @@ public void ApplyCapabilities(VSInternalServerCapabilities serverCapabilities, V
     }
 
     public async Task<WorkspaceEdit?> HandleRequestAsync(
-        VSInternalMapCodeParams request,
+        VSInternalMapCodeParams mapperParams,
         RazorRequestContext context,
         CancellationToken cancellationToken)
     {
         // TO-DO: Apply updates to the workspace before doing mapping. This is currently
         // unimplemented by the client, so we won't bother doing anything for now until
         // we determine what kinds of updates the client will actually send us.
-        Debug.Assert(request.Updates is null);
+        Debug.Assert(mapperParams.Updates is null);
 
-        if (request.Updates is not null)
+        if (mapperParams.Updates is not null)
         {
             return null;
         }
 
+        using var ts = TelemetryScope.Create(_telemetryReporter, "mapcode", Severity.Normal,
```

Comment:
Is this a specific telemetry event that is unified and used by all code mappers? If not, consider using our existing `_telemetryReporter.TrackLspRequest` pattern

Reply:
Done, and again in MapCode.cs to capture the sub-layer.